Transient IT test_snapshot_listing fails in anvilbox

https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/13044
```
FAIL: test_snapshot_listing (integration_test.IndexingIntegrationTest)
Test with a small page size to be sure paging works
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/azul_test_case.py", line 208, in wrapped
    return method(*args, **kwargs)
  File "/builds/ucsc/azul/test/integration_test.py", line 410, in test_snapshot_listing
    self.assertEqual(snapshots, paged_snapshots)
AssertionError: {'3ea[59586 chars]6', 'cbbf2f77-3830-435f-81e5-42bf8c8f764d': 'A[83202 chars]613'} != {'3ea[59586 chars]6', '84ca2098-ef4e-4d85-a988-424ccb7e33e6': 'A[83161 chars]613'}
Diff is 146780 characters long. Set self.maxDiff to None to see it.
----------------------------------------------------------------------
Ran 13 tests in 574.327s
FAILED (failures=1, skipped=2)
make: *** [Makefile:238: integration_test] Error 1
```

--- a/test/integration_test.py
+++ b/test/integration_test.py
@@ -391,11 +391,30 @@ class IndexingIntegrationTest(IntegrationTestCase, AlwaysTearDownTestCase):
         """
         Test with small page sizes to make sure paging works.
         """
+        # Without a filter, the test takes so long that there's a real risk of
+        # failure due to new snapshots being added mid-test.
+        snapshot_filters_by_deployment = {
+            'anvildev': 'anvil_',  # ~5 snapshots
+            'dev': 'hca_dev_5',  # ~10 snapshots
+            'anvilprod': 'anvil_page_',  # ~13 snapshots
+            'prod': '_dcp37'  # ~13 snapshots
+        }
+        filter = snapshot_filters_by_deployment[config.main_deployment_stage]
         for page_size in 1, 2:
             with self.subTest(page_size=page_size):
                 with mock.patch.object(TDRClient, 'page_size', page_size):
-                    paged_snapshots = self._public_tdr_client.snapshot_names_by_id()
-                snapshots = self._public_tdr_client.snapshot_names_by_id()
+                    paged_snapshots = self._tdr_client.snapshot_names_by_id(filter=filter)
+                snapshots = self._tdr_client.snapshot_names_by_id(filter=filter)
+                self.assertLess(len(snapshots), 20)
+                # Show that multiple pages were fetched, via the pigeonhole
+                # principle, and under the assumption that the TDR client
+                # correctly implements paging, and doesn't, for example, ignore
+                # the page size parameter. This test is designed to detect
+                # problems in the server-side implementation of paging, or our
+                # understanding of it. There is a unit test
+                # (test_list_snapshots_paging) dedicated to ensuring that the
+                # client-side implementation of paging is correct.
+                self.assertGreater(len(paged_snapshots), page_size)
                 self.assertEqual(snapshots, paged_snapshots)
 
     def test_indexing(self):

--- a/test/integration_test.py
+++ b/test/integration_test.py
@@ -389,13 +389,14 @@ class IndexingIntegrationTest(IntegrationTestCase, AlwaysTearDownTestCase):
 
     def test_snapshot_listing(self):
         """
-        Test with a small page size to be sure paging works
+        Test with small page sizes to make sure paging works.
         """
-        page_size = 5
-        with mock.patch.object(TDRClient, 'page_size', page_size):
-            paged_snapshots = self._public_tdr_client.snapshot_names_by_id()
-        snapshots = self._public_tdr_client.snapshot_names_by_id()
-        self.assertEqual(snapshots, paged_snapshots)
+        for page_size in 1, 2:
+            with self.subTest(page_size=page_size):
+                with mock.patch.object(TDRClient, 'page_size', page_size):
+                    paged_snapshots = self._public_tdr_client.snapshot_names_by_id()
+                snapshots = self._public_tdr_client.snapshot_names_by_id()
+                self.assertEqual(snapshots, paged_snapshots)
 
     def test_indexing(self):
 

